--- origsrc/qt-creator-1.1.0-src/src/libs/extensionsystem/patchedpluginloader.cpp	2009-04-21 09:13:12.000000000 -0500
+++ src/qt-creator-1.1.0-src/src/libs/extensionsystem/patchedpluginloader.cpp	2009-06-01 14:23:11.659281200 -0500
@@ -487,6 +487,7 @@ bool PatchedLibraryPrivate::loadPlugin()
     \table
     \header \i Platform \i Valid suffixes
     \row \i Windows     \i \c .dll
+    \row \i Cygwin      \i \c .dll
     \row \i Unix/Linux  \i \c .so
     \row \i AIX  \i \c .a
     \row \i HP-UX       \i \c .sl, \c .so (HP-UXi)
@@ -497,7 +498,7 @@ bool PatchedLibraryPrivate::loadPlugin()
  */
 bool QLibrary::isLibrary(const QString &fileName)
 {
-#if defined(Q_OS_WIN32) || defined(Q_OS_WINCE)
+#if defined(Q_OS_WIN32) || defined(Q_OS_WINCE) || defined(Q_OS_CYGWIN)
     return fileName.endsWith(QLatin1String(".dll"));
 #else
     QString completeSuffix = QFileInfo(fileName).completeSuffix();
@@ -907,6 +908,13 @@ bool PatchedLibraryPrivate::load_sys()
         }
 #elif defined(Q_OS_AIX)
         suffixes << ".a";
+#elif defined(Q_OS_CYGWIN)
+        prefixes << QLatin1String("cyg");
+        if (!fullVersion.isEmpty()) {
+            suffixes << QString::fromLatin1("-%1.dll").arg(fullVersion);
+        } else {
+            suffixes << QLatin1String(".dll");
+        }
 #else
         if (!fullVersion.isEmpty()) {
             suffixes << QString::fromLatin1(".so.%1").arg(fullVersion);
--- origsrc/qt-creator-1.1.0-src/src/libs/extensionsystem/pluginspec.cpp	2009-04-21 09:13:12.000000000 -0500
+++ src/qt-creator-1.1.0-src/src/libs/extensionsystem/pluginspec.cpp	2009-06-01 14:27:30.884108000 -0500
@@ -768,6 +768,8 @@ bool PluginSpecPrivate::loadLibrary()
     QString libName = QString("%1/%2.dll").arg(location).arg(name);
 #elif defined(Q_OS_MAC)
     QString libName = QString("%1/lib%2.dylib").arg(location).arg(name);
+#elif defined(Q_OS_CYGWIN)
+    QString libName = QString("%1/cyg%2.dll").arg(location).arg(name);
 #else
     QString libName = QString("%1/lib%2.so").arg(location).arg(name);
 #endif
@@ -778,6 +780,8 @@ bool PluginSpecPrivate::loadLibrary()
     QString libName = QString("%1/%2d.dll").arg(location).arg(name);
 #elif defined(Q_OS_MAC)
     QString libName = QString("%1/lib%2_debug.dylib").arg(location).arg(name);
+#elif defined(Q_OS_CYGWIN)
+    QString libName = QString("%1/cyg%2.dll").arg(location).arg(name);
 #else
     QString libName = QString("%1/lib%2.so").arg(location).arg(name);
 #endif
