--- origsrc/qt-creator-1.2.0-src/share/qtcreator/gdbmacros/gdbmacros.cpp	2009-06-29 01:20:00.731817200 -0500
+++ src/qt-creator-1.2.0-src/share/qtcreator/gdbmacros/gdbmacros.cpp	2009-06-29 03:02:32.606013300 -0500
@@ -68,6 +68,12 @@ int qtGhVersion = QT_VERSION;
 #include <set>
 #include <vector>
 
+#if defined(Q_OS_UNIX) && !defined(_GLIBCXX_USE_WCHAR_T)
+namespace std {
+	typedef basic_string<wchar_t> wstring;
+}
+#endif
+
 /*!
   \class QDumper
   \brief Helper class for producing "nice" output in Qt Creator's debugger.
