--- origsrc/qt-creator-2.1.0-src/src/libs/3rdparty/botan/src/alloc/alloc_mmap/mmap_mem.cpp	2011-02-11 11:29:12.000000000 -0600
+++ src/qt-creator-2.1.0-src/src/libs/3rdparty/botan/src/alloc/alloc_mmap/mmap_mem.cpp	2011-03-27 03:06:49.685819300 -0500
@@ -4,6 +4,7 @@
 *
 * Distributed under the terms of the Botan license
 */
+#undef __STRICT_ANSI__
 
 #include <botan/mmap_mem.h>
 #include <cstring>
--- origsrc/qt-creator-2.1.0-src/src/libs/3rdparty/botan/src/src.pro	2011-02-11 11:29:12.000000000 -0600
+++ src/qt-creator-2.1.0-src/src/libs/3rdparty/botan/src/src.pro	2011-03-27 03:11:44.105132700 -0500
@@ -284,7 +284,7 @@ unix {
            timer/gettimeofday/tm_unix.h
 }
 
-linux*-g++* {
+cygwin*|linux*-g++* {
     HEADERS += timer/posix_rt/tm_posix.h
 }
 
@@ -568,7 +568,7 @@ unix {
            timer/gettimeofday/tm_unix.cpp
 }
 
-linux* {
+cygwin*|linux* {
     SOURCES += timer/posix_rt/tm_posix.cpp
 }
 
