--- origsrc/qt-creator-2.1.0-src/src/libs/qmljs/qmljscomponentversion.h	2011-02-11 11:29:12.000000000 -0600
+++ src/qt-creator-2.1.0-src/src/libs/qmljs/qmljscomponentversion.h	2011-03-25 03:25:28.833306900 -0500
@@ -36,6 +36,9 @@
 
 #include "qmljs_global.h"
 
+#undef major
+#undef minor
+
 namespace QmlJS {
 
 class QMLJS_EXPORT ComponentVersion
--- origsrc/qt-creator-2.1.0-src/src/plugins/debugger/outputcollector.cpp	2011-02-11 11:29:12.000000000 -0600
+++ src/qt-creator-2.1.0-src/src/plugins/debugger/outputcollector.cpp	2011-03-25 03:28:13.413593800 -0500
@@ -51,6 +51,7 @@
 #include <sys/ioctl.h>
 #include <sys/types.h>
 #include <sys/stat.h>
+#include <sys/socket.h>
 #ifdef Q_OS_SOLARIS
 # include <sys/filio.h> // FIONREAD
 #endif
--- origsrc/qt-creator-2.1.0-src/src/shared/symbianutils/trkdevice.cpp	2011-02-11 11:29:12.000000000 -0600
+++ src/qt-creator-2.1.0-src/src/shared/symbianutils/trkdevice.cpp	2011-03-25 03:29:01.532077500 -0500
@@ -54,6 +54,7 @@
 
 #  include <stdio.h>
 #  include <sys/ioctl.h>
+#  include <sys/socket.h>
 #  include <sys/types.h>
 #  include <termios.h>
 #  include <errno.h>
