--- origsrc/qt-creator-2.2.1-src/src/plugins/qt4projectmanager/qtversionmanager.cpp	2011-06-15 14:47:14.000000000 -0500
+++ src/qt-creator-2.2.1-src/src/plugins/qt4projectmanager/qtversionmanager.cpp	2011-07-13 03:20:54.910068600 -0500
@@ -1702,6 +1702,9 @@ QString QtVersion::qtCorePath() const
             if (info.isReadable()) {
                 if (file.startsWith(QLatin1String("libQtCore"))
                         || file.startsWith(QLatin1String("QtCore"))) {
+#ifdef Q_OS_CYGWIN
+                    return QString::fromLatin1("/usr/bin/cygQtCore-4.dll");
+#endif
                     // Only handle static libs if we can not find dynamic ones:
                     if (file.endsWith(".a") || file.endsWith(".lib"))
                         staticLibs.append(info);
