--- origsrc/qt-creator-2.2.1-src/share/qtcreator/static.pro	2011-06-15 14:47:14.000000000 -0500
+++ src/qt-creator-2.2.1-src/share/qtcreator/static.pro	2011-07-13 02:44:44.850833500 -0500
@@ -80,7 +80,7 @@ defineReplace(stripSrcResourceDir) {
     win32 {
         !contains(1, ^.:.*):1 = $$OUT_PWD/$$1
     } else {
-        !contains(1, ^/.*):1 = $$OUT_PWD/$$1
+        !contains(1, ^/.*):1 = /$$OUT_PWD/$$1
     }
     out = $$cleanPath($$1)
     out ~= s|^$$re_escape($$SRCRESOURCEDIR)||$$i_flag
