--- origsrc/qt-creator-2.5.0-src/share/qtcreator/static.pro	2012-05-09 07:13:19.000000000 -0500
+++ src/qt-creator-2.5.0-src/share/qtcreator/static.pro	2012-05-13 22:34:32.783354800 -0500
@@ -82,7 +82,7 @@ defineReplace(stripSrcResourceDir) {
     win32 {
         !contains(1, ^.:.*):1 = $$OUT_PWD/$$1
     } else {
-        !contains(1, ^/.*):1 = $$OUT_PWD/$$1
+        !contains(1, ^/.*):1 = /$$OUT_PWD/$$1
     }
     out = $$cleanPath($$1)
     out ~= s|^$$re_escape($$SRCRESOURCEDIR)||$$i_flag
--- origsrc/qt-creator-2.5.0-src/src/app/app.pro	2012-05-09 07:13:19.000000000 -0500
+++ src/qt-creator-2.5.0-src/src/app/app.pro	2012-05-13 16:22:38.979078200 -0500
@@ -40,1 +40,1 @@ OTHER_FILES += qtcreator.rc \
-QMAKE_SUBSTITUTES += $$PWD/app_version.h.in
+QMAKE_SUBSTITUTES += app_version.h.in
--- origsrc/qt-creator-2.5.0-src/src/qtcreatorplugin.pri	2012-05-09 07:13:19.000000000 -0500
+++ src/qt-creator-2.5.0-src/src/qtcreatorplugin.pri	2012-05-13 20:53:32.841745900 -0500
@@ -38,8 +38,8 @@ defineReplace(stripOutDir) {
     return($$1)
 }
 
-PLUGINSPEC = $$_PRO_FILE_PWD_/$${TARGET}.pluginspec
-PLUGINSPEC_IN = $${PLUGINSPEC}.in
+PLUGINSPEC = $$OUT_PWD/$${TARGET}.pluginspec
+PLUGINSPEC_IN = $$_PRO_FILE_PWD_/$${TARGET}.pluginspec.in
 exists($$PLUGINSPEC_IN) {
     OTHER_FILES += $$PLUGINSPEC_IN
     QMAKE_SUBSTITUTES += $$PLUGINSPEC_IN
