--- origsrc/qt-creator-2.8.1-src/src/plugins/qtsupport/baseqtversion.cpp	2013-09-10 21:55:34.827889600 -0500
+++ src/qt-creator-2.8.1-src/src/plugins/qtsupport/baseqtversion.cpp	2013-09-10 21:54:24.709879100 -0500
@@ -1509,6 +1509,9 @@ FileName BaseQtVersion::qtCorePath(const
     QStringList dirs;
     dirs << qmakeProperty(versionInfo, "QT_INSTALL_LIBS")
          << qmakeProperty(versionInfo, "QT_INSTALL_BINS");
+#ifdef Q_OS_CYGWIN
+    dirs << QString::fromLatin1("/usr/bin");
+#endif
 
     QFileInfoList staticLibs;
     foreach (const QString &dir, dirs) {
@@ -1529,6 +1532,10 @@ FileName BaseQtVersion::qtCorePath(const
             if (info.isReadable()) {
                 if (file.startsWith(QLatin1String("libQtCore"))
                         || file.startsWith(QLatin1String("libQt5Core"))
+#ifdef Q_OS_CYGWIN
+                        || file.startsWith(QLatin1String("cygQtCore"))
+                        || file.startsWith(QLatin1String("cygQt5Core"))
+#endif
                         || file.startsWith(QLatin1String("QtCore"))
                         || file.startsWith(QLatin1String("Qt5Core"))) {
                     // Only handle static libs if we can not find dynamic ones:
